### PR TITLE
Encrypted ami

### PR DESCRIPTION
**Pull request by savagegus**
Copied and modified from the public ami plugin.

### CODE REVIEW

**Comment by savagegus** [comment]
@matthewdfuller what is the process to review this?

**Comment by matthewdfuller** [comment]
@savagegus - thanks so much for the PR! We'll take a look shortly and see if and other changes are needed before merging.

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "node",
+            "request": "launch",
+            "name": "Launch Program",
+            "program": "${workspaceRoot}/index.js"
+        }
+    ]
+}
```
Comment (matthewdfuller) [inline .vscode/launch.json]: This file may have been accidentally committed - could you add it to the `.gitignore` instead?

File: plugins/ec2/encryptedAmi.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			for (i in describeImages.data) {
+				var image = describeImages.data[i];
+				for (j in image.BlockDeviceMappings) {
+					var volume = image.BlockDeviceMappings[j];
+					if (volume.hasOwnProperty('Ebs')) {
+						if (!volume.Ebs.Encrypted) {
+							found = true;
+							helpers.addResult(results, 2, 'AMI EBS volume is unencrypted', region, image.ImageId);
+						}
+					}
```
Comment (matthewdfuller) [inline plugins/ec2/encryptedAmi.js]: For this section, there may be a risk of adding the same result multiple times if the image has multiple `BlockDeviceMappings`. After the `helpers.addResult` line, it should `break` out of the inner for loop to move on to the next image.

File: .vscode/launch.json (hunk 1 above)
Comment (savagegus) [inline .vscode/launch.json]: yup, my bad

File: plugins/ec2/encryptedAmi.js (hunk 2 above)
Comment (savagegus) [inline plugins/ec2/encryptedAmi.js]: done

**Comment by savagegus** [comment]
Ok should be set.

**Comment by matthewdfuller** [comment]
@mattfinlayson - Apologies for the (very late) ping on this PR, but performing an evaluation of code contributed to CloudSploit and realized we don't have a license agreement in place for you. Would you mind using the below link to review and digitally-sign CloudSploit's CLA? Thanks in advanced!

**Comment by matthewdfuller** [comment]
Hi @mattfinlayson - following up on the above message to see if you could sign the attached [CLA](https://cla-assistant.io/cloudsploit/scans). Much appreciated!

**Comment by mattfinlayson** [comment]
Hi Matthew, unfortunately I did this at work, so I've sent the request over to my legal team. I don't think there will be an issue but there will be a delay.

**Comment by matthewdfuller** [comment]
Hi @mattfinlayson - no worries, thank you for the update and working to get this in place! Let me know what you hear back.

**Pull request merged**